tools/shared.py

```diff
@@ -471,6 +471,10 @@ def find_temp_directory():
 def get_emscripten_version(path):
   return open(path).read().strip().replace('"', '')
 
+# Returns true if Emscripten is running in 'strict' mode, in which deprecated compiler features are not supported.
+def is_emscripten_strict():
+  return (os.environ.get('EMCC_STRICT') and int(os.environ.get('EMCC_STRICT')) != 0) or Settings.STRICT
```

Comment:
In particular, we should have tests that check usage with both the env var and without, of things both working and not, so we hit that code path.